File: Phase 2/backend/api/model/resource.go
package model

import (
	"github.com/lib/pq"
	"gorm.io/gorm"
)

type Resource struct {
	gorm.Model
	Topic           pq.StringArray `gorm:"type:topic_enum[]" json:"topic"`
	Title           string         `json:"title"`
	Link            string         `json:"link"`
	Level           string         `json:"level"`
	Description     string         `json:"description"`
	LinkDescription string         `json:"link_description"`
}

File: Phase 2/frontend/lib/api/resource.ts
import { ApiResponse } from './auth';
import axios from './axios';

export interface Resource {
    id: number;
    title: string;
    description: string;
    link: string;
    topic: string[];
    level: string;
    createdAt: string;
}

const ResourceAPI = {
    getResources: (params?: { search?: string; level?: string }) => {
        const queryParams = new URLSearchParams();

        if (params?.search) queryParams.append('search', params.search);
        if (params?.level) queryParams.append('level', params.level);

        const queryString = queryParams.toString() ? `?${queryParams.toString()}` : '';

        return axios.get<ApiResponse<Resource[]>>(`/resources${queryString}`, {
            withCredentials: true,
        });
    },
};

export default ResourceAPI;
